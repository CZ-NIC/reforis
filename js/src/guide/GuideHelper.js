/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";

import { HELP_CONTENT } from "./constants";
import GuideControls from "./GuideControls/GuideControls";

GuideHelper.propTypes = {
    ws: PropTypes.object.isRequired,
    workflow: PropTypes.string.isRequired,
    step: PropTypes.string.isRequired,
    next_step: PropTypes.string.isRequired,
    completed: PropTypes.bool,
};

GuideHelper.defaultProps = {
    completed: false,
};

export default function GuideHelper({
    ws, workflow, step, next_step, completed,
}) {
    const stepContent = HELP_CONTENT[workflow][step];
    if (!stepContent) {
        return null;
    }

    const isCompletedVisible = stepContent && completed && stepContent.completed;
    if (!(stepContent.initial || isCompletedVisible)) {
        return null;
    }

    return (
        <div className="card guide-card">
            <div className="card-body">
                {stepContent.initial && <ParagraphsArray content={stepContent.initial} />}
                {isCompletedVisible && <p className="font-weight-bold">{ stepContent.completed }</p>}
            </div>
            <GuideControls ws={ws} next_step={next_step} />
        </div>
    );
}

ParagraphsArray.propTypes = {
    content: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.arrayOf(PropTypes.string),
    ]),
};

function ParagraphsArray({ content }) {
    return Array.isArray(content)
        // eslint-disable-next-line react/no-array-index-key
        ? content.map((line, index) => <p key={index} dangerouslySetInnerHTML={{ __html: line }} />)
        : <p dangerouslySetInnerHTML={{ __html: content }} />;
}
