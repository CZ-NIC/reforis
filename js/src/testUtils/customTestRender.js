/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

/* eslint import/export: "off" */

import React from 'react';
import {UIDReset} from 'react-uid';
import {StaticRouter} from 'react-router';
import {render} from '@testing-library/react'

function Wrapper({children}) {
    return <StaticRouter>
        <UIDReset>
            {children}
        </UIDReset>
    </StaticRouter>
}

const customTestRender = (ui, options) => render(ui, {wrapper: Wrapper, ...options});

// re-export everything
export * from '@testing-library/react'

// override render method
export {customTestRender as render}
