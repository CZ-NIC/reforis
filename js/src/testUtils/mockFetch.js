/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

export default function mockFetch(data) {
    return jest.fn(() =>
        new Promise((resolve, reject) => {
            resolve({
                ok: true,
                json: () => {
                    return data ? data : {};
                },
            });
        })
    );
}
