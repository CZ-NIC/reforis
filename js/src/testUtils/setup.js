/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import mockAxios from "jest-mock-axios";
import moment from "moment-timezone";

// Setup axios cleanup
global.afterEach(() => {
    mockAxios.reset();
});

jest.doMock('moment', () => {
    moment.tz.setDefault('UTC');
    return moment;
});

jest.doMock("utils/vfs_fonts", () => ({}));

window.WebSocket = jest.fn();
