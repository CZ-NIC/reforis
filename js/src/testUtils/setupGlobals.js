/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

// Fake babel (gettext) used for docs
global._ = str => str;
global.babel = {format: (str) => str};
global.ForisTranslations = {};
