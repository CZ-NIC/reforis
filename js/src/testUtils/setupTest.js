/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import mockAxios from 'jest-mock-axios';
import moment from 'moment-timezone';

// Setup axios cleanup
global.afterEach(() => {
    mockAxios.reset();
});

// Mock babel (gettext)
window._ = str => str;
window.babel = {format: (str) => str};
window.ForisTranslations = {};

// Mock scrollIntoView
window.HTMLElement.prototype.scrollIntoView = () => {
};

jest.doMock('moment', () => {
    moment.tz.setDefault('UTC');
    return moment;
});

Date.now = jest.fn(() => new Date(Date.UTC(2019, 1, 1, 12, 13, 14)).valueOf());

jest.doMock('utils/vfs_fonts', () => {
    return {};
});
