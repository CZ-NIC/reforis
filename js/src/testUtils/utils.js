/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */


export function createPortalContainer(id) {
    let portalRoot = document.getElementById(id);
        if (!portalRoot) {
            portalRoot = document.createElement('div');
            portalRoot.setAttribute('id', id);
            document.body.appendChild(portalRoot)
        }
}
