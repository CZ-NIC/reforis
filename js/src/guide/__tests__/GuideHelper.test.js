/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { render } from "foris/testUtils/customTestRender";
import GuideHelper from "../GuideHelper";
import { WebSockets } from "foris";

describe("<GuideHelp/>", () => {
    const webSockets = new WebSockets();

    it("displays lan help without completed message", () => {
        const { container } = render(
            <GuideHelper
                ws={webSockets}
                workflow="router"
                step="lan"
                next_step="some"
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("displays lan help with completed message", () => {
        const { container } = render(
            <GuideHelper
                ws={webSockets}
                workflow="router"
                step="lan"
                next_step="some"
                completed
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("displays lan help without completed message - server workflow", () => {
        const { container } = render(
            <GuideHelper
                ws={webSockets}
                workflow="bridge"
                step="lan"
                next_step="some"
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("doesn't display completed message if it doesn't exist", () => {
        const { container } = render(
            <GuideHelper
                ws={webSockets}
                workflow="bridge"
                step="finished"
                next_step="some"
                completed
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("doesn't display anything if step doesn't exist", () => {
        const { container } = render(
            <GuideHelper
                ws={webSockets}
                workflow="router"
                step="qwe"
                next_step="some"
                completed
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    it("doesn't display initial message if it doesn't exist", () => {
        const { container } = render(
            <GuideHelper
                ws={webSockets}
                workflow="router"
                step="profile"
                next_step="some"
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
