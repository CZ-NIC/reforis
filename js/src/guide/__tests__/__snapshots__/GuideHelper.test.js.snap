// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GuideHelp/> displays lan help with completed message 1`] = `
<div
  class="card guide-card"
>
  <div
    class="card-body"
  >
    <p>
      Now you should configure your LAN interface. Note that when you change your network settings you probably won't be able to connect to the configuration interface unless you restart the network on your current device.
    </p>
    <p
      class="font-weight-bold"
    >
      You've configured your LAN interface. Try to test whether settings work properly and if so you can safely proceed to the next step.
    </p>
  </div>
  <div
    class="guide-controls"
  >
    <div
      class="dropdown"
    >
      <button
        class="nav-item btn guide-controls-button btn-primary"
        type="button"
      >
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </button>
      <div
        class="dropdown-menu"
        id="languages-dropdown-menu"
      >
        <div
          class="dropdown-header"
        >
          <h5>
            Languages
          </h5>
        </div>
        <div
          class="dropdown-divider"
        />
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </div>
    </div>
    <button
      class="btn guide-controls-button btn btn-warning"
      type="button"
    >
       
       
      <div
        class="d-none d-sm-block"
      >
        Skip guide
         
      </div>
      <i
        class="fas fa-forward"
      />
    </button>
    <a
      class="guide-controls-button btn btn-primary blinking"
      href="/some"
    >
      <div
        class="d-none d-sm-block"
      >
        Next step
      </div>
      <i
        class="fas fa-step-forward"
      />
    </a>
  </div>
</div>
`;

exports[`<GuideHelp/> displays lan help without completed message - server workflow 1`] = `
<div
  class="card guide-card"
>
  <div
    class="card-body"
  >
    <p>
      To act as a local server, there's no need to manage LAN (if you still want to manage it reset the guide and choose the Router workflow). You probably want to act as a client here thus select 
      <strong>
        Computer
      </strong>
       mode here.
    </p>
    <p>
      If you select the 
      <strong>
        static
      </strong>
       configuration be sure that the IP addresses are entered correctly otherwise you won't be able to access this configuration interface when the new settings are applied.
    </p>
    <p>
      If you select the 
      <strong>
        DHCP
      </strong>
       method you probably you need to obtain a new IP address of this device. You can obtain it from the DHCP server which is managing your LAN. Then you need to connect to the new IP address to proceed the guide.
    </p>
    <p>
      Note that either way you might need to re-plug your ethernet cables after you update the settings here.
    </p>
  </div>
  <div
    class="guide-controls"
  >
    <div
      class="dropdown"
    >
      <button
        class="nav-item btn guide-controls-button btn-primary"
        type="button"
      >
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </button>
      <div
        class="dropdown-menu"
        id="languages-dropdown-menu"
      >
        <div
          class="dropdown-header"
        >
          <h5>
            Languages
          </h5>
        </div>
        <div
          class="dropdown-divider"
        />
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </div>
    </div>
    <button
      class="btn guide-controls-button btn btn-warning"
      type="button"
    >
       
       
      <div
        class="d-none d-sm-block"
      >
        Skip guide
         
      </div>
      <i
        class="fas fa-forward"
      />
    </button>
    <a
      class="guide-controls-button btn btn-primary blinking"
      href="/some"
    >
      <div
        class="d-none d-sm-block"
      >
        Next step
      </div>
      <i
        class="fas fa-step-forward"
      />
    </a>
  </div>
</div>
`;

exports[`<GuideHelp/> displays lan help without completed message 1`] = `
<div
  class="card guide-card"
>
  <div
    class="card-body"
  >
    <p>
      Now you should configure your LAN interface. Note that when you change your network settings you probably won't be able to connect to the configuration interface unless you restart the network on your current device.
    </p>
  </div>
  <div
    class="guide-controls"
  >
    <div
      class="dropdown"
    >
      <button
        class="nav-item btn guide-controls-button btn-primary"
        type="button"
      >
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </button>
      <div
        class="dropdown-menu"
        id="languages-dropdown-menu"
      >
        <div
          class="dropdown-header"
        >
          <h5>
            Languages
          </h5>
        </div>
        <div
          class="dropdown-divider"
        />
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </div>
    </div>
    <button
      class="btn guide-controls-button btn btn-warning"
      type="button"
    >
       
       
      <div
        class="d-none d-sm-block"
      >
        Skip guide
         
      </div>
      <i
        class="fas fa-forward"
      />
    </button>
    <a
      class="guide-controls-button btn btn-primary blinking"
      href="/some"
    >
      <div
        class="d-none d-sm-block"
      >
        Next step
      </div>
      <i
        class="fas fa-step-forward"
      />
    </a>
  </div>
</div>
`;

exports[`<GuideHelp/> doesn't display anything if step doesn't exist 1`] = `null`;

exports[`<GuideHelp/> doesn't display completed message if it doesn't exist 1`] = `
<div
  class="card guide-card"
>
  <div
    class="card-body"
  >
    <p>
      Congratulations! Now your device should be able to act as a server on your local network.
    </p>
  </div>
  <div
    class="guide-controls"
  >
    <div
      class="dropdown"
    >
      <button
        class="nav-item btn guide-controls-button btn-primary"
        type="button"
      >
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </button>
      <div
        class="dropdown-menu"
        id="languages-dropdown-menu"
      >
        <div
          class="dropdown-header"
        >
          <h5>
            Languages
          </h5>
        </div>
        <div
          class="dropdown-divider"
        />
        <div
          class="spinner-border spinner-border-sm"
          role="status"
        >
          <span
            class="sr-only"
          />
        </div>
      </div>
    </div>
    <button
      class="btn guide-controls-button btn btn-warning"
      type="button"
    >
       
       
      <div
        class="d-none d-sm-block"
      >
        Skip guide
         
      </div>
      <i
        class="fas fa-forward"
      />
    </button>
    <a
      class="guide-controls-button btn btn-primary blinking"
      href="/some"
    >
      <div
        class="d-none d-sm-block"
      >
        Next step
      </div>
      <i
        class="fas fa-step-forward"
      />
    </a>
  </div>
</div>
`;

exports[`<GuideHelp/> doesn't display initial message if it doesn't exist 1`] = `null`;
